internal/services/devcenter/dev_center_environment_type_resource_test.go

```diff
@@ -0,0 +1,180 @@
+package devcenter_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/go-azure-sdk/resource-manager/devcenter/2023-04-01/environmenttypes"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance/check"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type DevCenterEnvironmentTypeTestResource struct{}
+
+func TestAccDevCenterEnvironmentType_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_dev_center_environment_type", "test")
+	r := DevCenterEnvironmentTypeTestResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccDevCenterEnvironmentType_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_dev_center_environment_type", "test")
+	r := DevCenterEnvironmentTypeTestResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.RequiresImportErrorStep(r.requiresImport),
+	})
+}
+
+func TestAccDevCenterEnvironmentType_complete(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_dev_center_environment_type", "test")
+	r := DevCenterEnvironmentTypeTestResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.complete(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccDevCenterEnvironmentType_update(t *testing.T) {
```

Comment:
please use the following the steps to also test add/remove tags:

basic -> complete -> update -> basic

Reply:
Updated